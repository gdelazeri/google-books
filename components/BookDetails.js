import React from 'react';
import {
  Text,
  View,
  Image,
  Dimensions,
  Linking,
} from 'react-native';
import { Icon } from 'react-native-elements';
import Styles from '../constants/Styles';
import Colors from '../constants/Colors';
import CustomButton from '../components/CustomButton';

// Largura da tela
const { width } = Dimensions.get('window');

export default class BookDetails extends React.Component {
  openLink(url) {
    Linking.canOpenURL(url).then(supported => {
      if (supported) {
        Linking.openURL(url);
      }
    });
  }

  render() {
    const book = this.props.book;
    let image;
    if (book.volumeInfo.imageLinks) {
      if (book.volumeInfo.imageLinks.thumbnail) image = { uri: book.volumeInfo.imageLinks.thumbnail };
      else if (book.volumeInfo.imageLinks.smallThumbnail) image = { uri: book.volumeInfo.imageLinks.smallThumbnail };
      else if (book.volumeInfo.imageLinks.small) image = { uri: book.volumeInfo.imageLinks.small };
      else if (book.volumeInfo.imageLinks.medium) image = { uri: book.volumeInfo.imageLinks.medium };
      else if (book.volumeInfo.imageLinks.large) image = { uri: book.volumeInfo.imageLinks.large };
      else if (book.volumeInfo.imageLinks.extraLarge) image = { uri: book.volumeInfo.imageLinks.extraLarge };
    }
    return <View>
      <Text style={[Styles.text18, Styles.textBold]}>Mais detalhes</Text>
      <View style={Styles.viewDivider} />
      <View style={Styles.inline}>
        {image && <Image source={image} style={{ width: width*0.3, height: width*0.45 }} />}
        <View style={{ width: 15 }} />
        <View>
          {book.saleInfo.saleability === 'NOT_FOR_SALE' && <View>
            <Text style={[Styles.text16, Styles.textBold]}>Não disponível para venda</Text>
            <View style={Styles.viewDivider5} />
          </View>}
          {book.saleInfo.saleability === 'FOR_SALE' && <View>
            <Text style={Styles.text16}>
              <Text style={Styles.textBold}>Preço: </Text>
              {book.saleInfo.listPrice.amount.toString().replace('.',',')} {book.saleInfo.listPrice.currencyCode}
            </Text>
            <View style={Styles.viewDivider5} />
          </View>}
          {book.volumeInfo.pageCount && <View>
            <Text style={Styles.text16}>
              <Text style={Styles.textBold}>Páginas: </Text>
              {book.volumeInfo.pageCount}
            </Text>
            <View style={Styles.viewDivider5} />
          </View>}
          {book.volumeInfo.averageRating !== undefined && <View style={Styles.inline}>
            <Text style={Styles.text16}>
              <Text style={Styles.textBold}>Avaliação: </Text>
              {book.volumeInfo.averageRating.toFixed(2).replace('.',',')}
            </Text>
            <View style={{ width: 2 }} />
            <Icon type='material-community' name='star' size={20} color={Colors.lightText} />
            <View style={Styles.viewDivider5} />
          </View>}
          {book.saleInfo.buyLink && <View style={[Styles.center, { width: (width*0.55)-15 }]}>
            <View style={Styles.viewDivider15} />
            <CustomButton
              width={width*0.4}
              buttonStyle={Styles.btnBorderPrimary}
              textStyle={{ color: Colors.primaryColor }}
              onPress={() => this.openLink(book.saleInfo.buyLink)}
              title='COMPRAR'
            />
          </View>}
        </View>
      </View>
    </View>
  }
}
