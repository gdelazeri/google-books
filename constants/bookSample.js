const bookSample = {
    "kind": "books#volume",
    "id": "aMT3cHZPwUcC",
    "etag": "et5vsPZRyzc",
    "selfLink": "https://www.googleapis.com/books/v1/volumes/aMT3cHZPwUcC",
    "volumeInfo": {
     "title": "Cooking for Harry",
     "subtitle": "A Low-Carbohydrate Novel",
     "authors": [
      "Kay-Marie James"
     ],
     "publisher": "Ballantine Books",
     "publishedDate": "2004-02-17",
     "description": "Harry became a fabulous cook. It began with a simple indulgence: secret bowls of buttery popcorn that he and his wife, Francie, would share after the children were tucked into bed. The aroma of melting butter, the hot kernels on their tongues, the salt crystals sticking to their lips—it was their own private romantic feast, imbuing their marriage with a new kind of passion. Soon, Harry began to dazzle Francie with luscious bisques and brioches, delectable soufflés, rich risottos, and classic versions of coq au vin that left her breathless. Their family life came to revolve around the dinner table, where each night Harry’s cooking brought Francie and their four children together for an awe-inspiring and mouthwatering meal. But inevitably the years slip by, and when all but one child has left the house, Harry wins a digital scale in his company’s Holiday Raffle and their happy bubble bursts in a single instant. Harry’s cooking has finally caught up with him. His doctor confirms it: He desperately needs to lose weight. Terrified of losing him, Francie puts Harry on a strict diet, leaving him eternally frustrated at the table and in the kitchen. When they both realize that he has to take a break from his culinary passions if this diet is to work, Francie begins to cook. Eventually a younger-looking, leaner, and more driven Harry emerges—one so newly committed to his job and his low-carb support group that not only is he no longer in the kitchen, he’s hardly ever at home. Feeling confused by the dynamics of their new relationship, Francie must contend with her need to keep Harry on his diet, and also with the women who have suddenly begun to eye her truly attractive husband. The question now becomes: Will love be enough to keep this marriage together, or will the Atkins Diet ultimately tear Harry and Francie apart? Pop a pan of cookies into the oven and put up your feet. Cooking for Harry is a deliciously good time. From the Hardcover edition.",
     "industryIdentifiers": [
      {
       "type": "ISBN_13",
       "identifier": "9781400080731"
      },
      {
       "type": "ISBN_10",
       "identifier": "1400080738"
      }
     ],
     "readingModes": {
      "text": true,
      "image": true
     },
     "pageCount": 224,
     "printType": "BOOK",
     "categories": [
      "Fiction"
     ],
     "averageRating": 4.5,
     "ratingsCount": 2,
     "maturityRating": "NOT_MATURE",
     "allowAnonLogging": false,
     "contentVersion": "0.1.0.0.preview.3",
     "panelizationSummary": {
      "containsEpubBubbles": false,
      "containsImageBubbles": false
     },
     "imageLinks": {
      "smallThumbnail": "http://books.google.com/books/content?id=aMT3cHZPwUcC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
      "thumbnail": "http://books.google.com/books/content?id=aMT3cHZPwUcC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
     },
     "language": "en",
     "previewLink": "http://books.google.com.br/books?id=aMT3cHZPwUcC&pg=PT86&dq=harry&hl=&cd=51&source=gbs_api",
     "infoLink": "https://play.google.com/store/books/details?id=aMT3cHZPwUcC&source=gbs_api",
     "canonicalVolumeLink": "https://play.google.com/store/books/details?id=aMT3cHZPwUcC"
    },
    "saleInfo": {
     "country": "BR",
     "saleability": "FOR_SALE",
     "isEbook": true,
     "listPrice": {
      "amount": 24.66,
      "currencyCode": "BRL"
     },
     "retailPrice": {
      "amount": 24.66,
      "currencyCode": "BRL"
     },
     "buyLink": "https://play.google.com/store/books/details?id=aMT3cHZPwUcC&rdid=book-aMT3cHZPwUcC&rdot=1&source=gbs_api",
     "offers": [
      {
       "finskyOfferType": 1,
       "listPrice": {
        "amountInMicros": 2.466E7,
        "currencyCode": "BRL"
       },
       "retailPrice": {
        "amountInMicros": 2.466E7,
        "currencyCode": "BRL"
       },
       "giftable": true
      }
     ]
    },
    "accessInfo": {
     "country": "BR",
     "viewability": "PARTIAL",
     "embeddable": true,
     "publicDomain": false,
     "textToSpeechPermission": "ALLOWED_FOR_ACCESSIBILITY",
     "epub": {
      "isAvailable": true,
      "acsTokenLink": "http://books.google.com.br/books/download/Cooking_for_Harry-sample-epub.acsm?id=aMT3cHZPwUcC&format=epub&output=acs4_fulfillment_token&dl_type=sample&source=gbs_api"
     },
     "pdf": {
      "isAvailable": true,
      "acsTokenLink": "http://books.google.com.br/books/download/Cooking_for_Harry-sample-pdf.acsm?id=aMT3cHZPwUcC&format=pdf&output=acs4_fulfillment_token&dl_type=sample&source=gbs_api"
     },
     "webReaderLink": "http://play.google.com/books/reader?id=aMT3cHZPwUcC&hl=&printsec=frontcover&source=gbs_api",
     "accessViewStatus": "SAMPLE",
     "quoteSharingAllowed": false
    },
    "searchInfo": {
     "textSnippet": "“It&#39;s a long shot,” \u003cb\u003eHarry\u003c/b\u003e said. “But if this happens, our lives could really change.” \u003cbr\u003e\nHe didn&#39;t seem to realize that our lives had already changed. Several times, I took \u003cbr\u003e\nLucy&#39;s business card out of my coat pocket. Call me if you ever want to talk."
    }
};

export default bookSample;