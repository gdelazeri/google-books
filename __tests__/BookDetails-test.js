import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import BookDetails from '../components/BookDetails';
import bookSample from '../constants/bookSample';

describe('BookDetails', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`renders the BookDetails component`, () => {
    const tree = renderer.create(<BookDetails book={bookSample} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
