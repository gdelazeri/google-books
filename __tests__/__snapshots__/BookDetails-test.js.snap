// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetails renders the BookDetails component 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#707d8a",
          "fontFamily": "open-sans",
          "fontSize": 18,
        },
        Object {
          "color": "#707d8a",
          "fontFamily": "open-sans-bold",
        },
      ]
    }
  >
    Mais detalhes
  </Text>
  <View
    style={
      Object {
        "height": 10,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://books.google.com/books/content?id=aMT3cHZPwUcC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
        }
      }
      style={
        Object {
          "height": 337.5,
          "width": 225,
        }
      }
    />
    <View
      style={
        Object {
          "width": 15,
        }
      }
    />
    <View>
      <View>
        <Text
          style={
            Object {
              "color": "#707d8a",
              "fontFamily": "open-sans",
              "fontSize": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#707d8a",
                "fontFamily": "open-sans-bold",
              }
            }
          >
            Preço: 
          </Text>
          24,66
           
          BRL
        </Text>
        <View
          style={
            Object {
              "height": 5,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#707d8a",
              "fontFamily": "open-sans",
              "fontSize": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#707d8a",
                "fontFamily": "open-sans-bold",
              }
            }
          >
            Páginas: 
          </Text>
          224
        </Text>
        <View
          style={
            Object {
              "height": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#707d8a",
              "fontFamily": "open-sans",
              "fontSize": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#707d8a",
                "fontFamily": "open-sans-bold",
              }
            }
          >
            Avaliação: 
          </Text>
          4,50
        </Text>
        <View
          style={
            Object {
              "width": 2,
            }
          }
        />
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 5,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "width": 397.50000000000006,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "height": 15,
            }
          }
        />
        <View
          style={
            Object {
              "borderRadius": 5,
              "height": 40,
              "marginLeft": 0,
              "marginRight": 0,
              "width": 300,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#1a73e8",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                  "padding": 8,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1a73e8",
                    "fontFamily": "open-sans-bold",
                    "fontSize": 16,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                COMPRAR
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
